src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+    fromProto tInfo = do
+        ti <- tInfo ^. Proto.maybe'tokenomics
+        case ti of
+            Proto.TokenomicsInfo'V0' v0 -> do
+                rsTotalAmount <- fromProto =<< v0 ^? ProtoFields.totalAmount
+                rsTotalEncryptedAmount <- fromProto =<< v0 ^? ProtoFields.totalEncryptedAmount
+                rsBakingRewardAccount <- fromProto =<< v0 ^? ProtoFields.bakingRewardAccount
+                rsFinalizationRewardAccount <- fromProto =<< v0 ^? ProtoFields.finalizationRewardAccount
+                rsGasAccount <- fromProto =<< v0 ^? ProtoFields.gasAccount
+                rsProtocolVersion <- fromProto =<< v0 ^? ProtoFields.protocolVersion
+                return QueryTypes.RewardStatusV0{..}
+            Proto.TokenomicsInfo'V1' v1 -> do
+                rsTotalAmount <- fromProto =<< v1 ^? ProtoFields.totalAmount
+                rsTotalEncryptedAmount <- fromProto =<< v1 ^? ProtoFields.totalEncryptedAmount
+                rsBakingRewardAccount <- fromProto =<< v1 ^? ProtoFields.bakingRewardAccount
+                rsFinalizationRewardAccount <- fromProto =<< v1 ^? ProtoFields.finalizationRewardAccount
+                rsGasAccount <- fromProto =<< v1 ^? ProtoFields.gasAccount
+                rsFoundationTransactionRewards <- fromProto =<< v1 ^? ProtoFields.foundationTransactionRewards
+                rsNextPaydayTime <- fmap timestampToUTCTime . fromProto =<< v1 ^? ProtoFields.nextPaydayTime
+                rsNextPaydayMintRate <- fromProto =<< v1 ^? ProtoFields.nextPaydayMintRate
+                rsTotalStakedCapital <- fromProto =<< v1 ^? ProtoFields.totalStakedCapital
+                rsProtocolVersion <- fromProto =<< v1 ^? ProtoFields.protocolVersion
+                return QueryTypes.RewardStatusV1{..}
+
+-- |Input for `invokeInstanceV2`.
+type InvokeInstanceInput = (BlockHashInput, ContractContext)
+
+instance ToProto InvokeInstanceInput where
+    type Output InvokeInstanceInput = Proto.InvokeInstanceRequest
+    toProto (bhi, ContractContext{..}) =
+        Proto.make $ do
+            ProtoFields.blockHash .= toProto bhi
+            ProtoFields.maybe'invoker .= fmap toProto ccInvoker
+            ProtoFields.instance' .= toProto ccContract
+            ProtoFields.amount .= toProto ccAmount
+            ProtoFields.entrypoint .= toProto ccMethod
+            ProtoFields.parameter .= toProto ccParameter
+            ProtoFields.energy .= toProto ccEnergy
+
+instance FromProto Proto.ContractEvent where
+    type Output' Proto.ContractEvent = Wasm.ContractEvent
+    -- FIXME: Can we just use deMkSerialize here? Types check out, but fromShort
+    -- is used in the other direction...
+    fromProto ce = return . Wasm.ContractEvent . BSS.toShort $ ce ^. ProtoFields.value
+
+instance FromProto Proto.Parameter where
+    type Output' Proto.Parameter = Wasm.Parameter
+    -- FIXME: There are protocol-dependent limits on the length. Can something
+    -- be done to enforce this here?
```

Comment:
No. Since everything is at protocol 5 now the limit is the more generous one.

Reply:
So the limit to be used is the V5 one. And it will only increase monotonically over time?

Reply:
I don't think we want to make any promises, but I assume that will be the case, yes ;)

Reply:
Nice. Thanks.